client/src/lib.rs

```diff
@@ -258,11 +347,32 @@ impl<V: vdaf::Client<16>> ClientBuilder<V> {
         } else {
             default_http_client()?
         };
-        let (leader_hpke_config, helper_hpke_config) = try_join!(
-            aggregator_hpke_config(&self.parameters, &Role::Leader, &http_client),
-            aggregator_hpke_config(&self.parameters, &Role::Helper, &http_client)
-        )?;
+        // TODO(#3159): store/fetch HPKE configurations in a cache-control aware persistent cache
+        let leader_hpke_config = match self.leader_hpke_config {
+            Some(config) => config,
+            None => aggregator_hpke_config(&self.parameters, &Role::Leader, &http_client).await?,
+        };
+        let helper_hpke_config = match self.helper_hpke_config {
+            Some(config) => config,
+            None => aggregator_hpke_config(&self.parameters, &Role::Helper, &http_client).await?,
+        };
+
+        #[cfg(feature = "ohttp")]
+        let ohttp_config = if let Some(ohttp_config) = self.ohttp_config {
```

Comment:
I'll give this a try.

Reply:
Done -- I tried adding the OHTTP key config future to the `try_join!`, but I can't work out how to add an extra future conditionally depending on a cargo feature, and I don't think it's worth the hassle of the gymnastics required to homogenize everything into a `Box<Future<Output = whatever>>` so it could go into `try_join_all`.